### PR TITLE
Heritage 319 histogram integration

### PR DESCRIPTION
**Pull request by JohnHeeryTNA**
Ticket URL:https://national-archives.atlassian.net/browse/HERITAGE-319

## About these changes

These are the three views for the Histogram using data from the API response (aggregations)
- by century
- within a century (by decade)
- within a decade (by year)

## How to check these changes

Run locally and test to see that the 3 views interact correctly, use 1900 as the initial century choice. 

Where possible, provide guidance to help your reviewer

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [x] Ensured that PR includes only commits relevant to the ticket
- [x] Waited for all CI jobs to pass before requesting a review
- [x] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/search/blocks/search_results_timeline_century.html [hunk 1]
```diff
@@ -1,129 +1,110 @@
...
-                }
-              };
-            })
-            .catch(error => {
-              console.error('Error fetching data:', error);
-              // Handle data fetching error (e.g., display an error message)
-            });
-
-  </script>
+  fetch('http://a08001c6c1d2e4c58bdf02589da33fca-0f850b505f873e8e.elb.eu-west-2.amazonaws.com/rosetta/data/search?aggs=decade,year,century&filter=group:community&filter=century:500&field=century:1900') // Replace with your server-side API endpoint
```
Comment (TNA-Allan) [inline templates/search/blocks/search_results_timeline_century.html:6]: Is it necessary to call the API exclusively? Also similarly called in other places.

File: templates/search/blocks/search_results_timeline_century_orig.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        const centuryIndex = Math.floor(century / 100);
+        centuryData[centuryIndex] = count;
+      }
+
+      return centuryData;
+    }
+
+    // Fetch the JSON data from a URL (if applicable)
+
+    fetch("/static/timeline/timeline.json")
```
Comment (TNA-Allan) [inline templates/search/blocks/search_results_timeline_century_orig.html:46]: Above is an exclusive API call, then is the static file necessary? Also called from other places.

File: templates/search/blocks/search_results_timeline_century.html (hunk 1 above)
Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century.html:6]: `&filter=century:500&field=century:1900`
One is filter, other is field - is this right?

Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century.html:6]: Could the JS be handled in a file of it own - for better readability and maintenance?

Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century.html:6]: The API domain is exposed in source. Should it be?

Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century.html:6]: Does the search term query, date filters, archive collection filters when applied also get applied to this API url?

File: templates/search/blocks/search_results_timeline_century_orig.html (hunk 2 above)
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_century_orig.html:46]: This is a back up file in case I need to revert, it will be removed eventually.  search_results_timeline_century_**orig.**html

File: templates/search/blocks/search_results_timeline_century.html (hunk 1 above)
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_century.html:6]: > Is it necessary to call the API exclusively? Also similarly called in other places.

The API gets called on page load for the three views.

Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_century.html:6]: > `&filter=century:500&field=century:1900` One is filter, other is field - is this right? Why are values constrained to 500, 1900?

I need to  check this with Lee (K-Int), we may not need these parameters at the moment

File: templates/search/blocks/search_results_timeline_per_century.html [hunk 3]
```diff
@@ -12,145 +12,119 @@
 
 <script>
 
-  const queryString = window.location.search;
-  const urlParams = new URLSearchParams(queryString);
-  //const startDate = urlParams.get('startDate');
-  const startDateString = urlParams.get('creation_date_from');
+  fetch('http://a08001c6c1d2e4c58bdf02589da33fca-0f850b505f873e8e.elb.eu-west-2.amazonaws.com/rosetta/data/search?aggs=decade,year,century&filter=group:community') // Replace with your server-side API endpoint
```
[the remark this answers is not in the document]
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_per_century.html:15]: I agree. We should have the js in a separate files(s) - these then will be compiled in the webpack process.

File: templates/search/blocks/search_results_timeline_century.html (hunk 1 above)
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_century.html:6]: > The API domain is exposed in source. Should it be?

By compiling the js via webpack it will de-exposed the api call.(see above comment).

Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_century.html:6]: > Does the search term query, date filters, archive collection filters when applied also get applied to this API url?

Not as yet but if we need to append these parameters to the url string, we can do via the js.

File: templates/search/blocks/search_results_timeline_per_century.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_results_timeline_per_century.html:15]: Not at this point. Currently we only need `aggs=decade,year,century&filter=group:community`.
We can add in additional parameters if we need to.

File: templates/search/blocks/search_results_timeline_century.html (hunk 1 above)
Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century.html:6]: > > The API domain is exposed in source. Should it be?
> 
> By compiling the js via webpack it will de-exposed the api call.(see above comment).

The API domain is exposed at GitHub source.

Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century.html:6]: > > Is it necessary to call the API exclusively? Also similarly called in other places.
> 
> The API gets called on page load for the three views.

What I meant was - why is the API called separately - can it not form a part of the same query for Timeline view?

File: templates/search/blocks/search_results_timeline_century_orig.html (hunk 2 above)
Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_century_orig.html:46]: > This is a back up file in case I need to revert, it will be removed eventually. search_results_timeline_century_**orig.**html

I think its easier as its a whole file to visit this file from history when its removed. Alternatively do you want to note a clean-up step in the ticket?

File: templates/search/blocks/search_results_timeline_per_century.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (TNA-Allan) [reply templates/search/blocks/search_results_timeline_per_century.html:15]: How does the scenario when user enters a search criteria fit into this?

File: scripts/src/timeline.js
Comment (TNA-Allan) [inline scripts/src/timeline.js:1]: - Why is the test alert in code?
- Is this file required?

File: scripts/src/modules/timeline/century.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
+fetch(
+    "http://a08001c6c1d2e4c58bdf02589da33fca-0f850b505f873e8e.elb.eu-west-2.amazonaws.com/rosetta/data/search?aggs=decade,year,century&filter=group:community",
+) // Replace with your server-side API endpoint
```
Comment (TNA-Allan) [inline scripts/src/modules/timeline/century.js:3]: Not able to view the chart on the test deployed branch.
Re: was loaded over HTTPS, but requested an insecure resource. This request has been blocked; the content must be served over HTTPS.

**Review by TNA-Allan — CHANGES_REQUESTED** [review]
Static data is visible in the repo https://github.com/nationalarchives/ds-ohos-wagtail/tree/feature/HERITAGE-319-histogram-integration/templates/static/timeline

**Comment by JohnHeeryTNA** [comment]
> Static data is visible in the repo https://github.com/nationalarchives/ds-ohos-wagtail/tree/feature/HERITAGE-319-histogram-integration/templates/static/timeline

/templates/static/timeline and static json files have now been removed.

File: scripts/src/timeline.js
Comment (JohnHeeryTNA) [reply scripts/src/timeline.js:1]: This file is required as this is the js that builds the initial timeline view.
alert code has been removed.

File: scripts/src/modules/timeline/century.js (hunk 4 above)
Comment (JohnHeeryTNA) [reply scripts/src/modules/timeline/century.js:3]: I'll check with Lee (K-int) on the cross domain request.

**Comment by JohnHeeryTNA** [comment]
Closing this PR - I'll set up a new one for recent changes.
